tests/test_graal.py

```diff
@@ -608,16 +608,17 @@ def test_backend_class(self):
     def test_gitpath_init(self):
         """Test gitpath initialization"""
 
+        parser = GraalCommand.setup_cmd_parser(CATEGORY_MOCKED)
```

Comment:
It seems that the change in `test_gitpath_init` prevents the execution of the `_pre_init` method (https://github.com/chaoss/grimoirelab-graal/blob/master/graal/graal.py#L459). In order to not decrease the test coverage, you could create a mocked GraalCommand class and change test in the following way:

```
class MockedGraalCommand(GraalCommand):
    BACKEND = MockedGraal

    @classmethod
    def setup_cmd_parser(cls):

        parser = GraalCommand.setup_cmd_parser(cls.BACKEND.CATEGORIES)

        return parser


....

def test_gitpath_init(self):
     """Test gitpath initialization"""

     args = ['http://example.com/',
                '--git-path', '/tmp/gitpath']

    cmd = MockedGraalCommand(*args)
    self.assertEqual(cmd.parsed_args.git_path, '/tmp/gitpath')

```

Reply:
Thanks for the suggestion, I was about to ask as I noticed the decrease in coverage in local run and couldn't figure out how to resolve it.
I'll work on this and update you in case i have any doubts.

Thanks

Reply:
Thank you @inishchith